javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -1113,13 +1226,37 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
 
     try {
       long start = logBeginProcess("Extracting " + file);
-      Integer loc = extractor.extract(f, state);
-      if (!extractor.getConfig().isExterns() && (loc == null || loc != 0)) seenCode = true;
+      ParseResultInfo loc = extractor.extract(f, state);
+      if (!extractor.getConfig().isExterns() && (loc == null || loc.getLinesOfCode() != 0)) seenCode = true;
       if (!extractor.getConfig().isExterns()) seenFiles = true;
+      for (ParseError err : loc.getParseErrors()) {
+        String msg = "A parse error occurred: " + err.getMessage() + ". Check the syntax of the file. If the file is invalid, correct the error or exclude the file from analysis.";
+        // file, relative to the source root
+        String relativeFilePath = file.toString();
+        if (relativeFilePath.startsWith(LGTM_SRC.toString())) {
+          relativeFilePath = relativeFilePath.substring(LGTM_SRC.toString().length() + 1);
+        }
```

Comment:
Let's avoid string manipulation when dealing with `Path`s.
```suggestion
        if (file.startsWith(LGTM_SRC)) {
          relativeFilePath = file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString();
        }
```